File: Week2/Homework/hw2-2/hw2-2a.js
db.grades.aggregate(
    {$match: {type: "homework"}},
    {$sort: { student_id:1, score:1}},
    {$group: {
        _id: "$student_id",    //unnecessary
        low: {$min: "$score"}, //unnecessary
        did: {$first: "$_id"}
    }})
    .result.forEach( function(e) {
            db.grades.remove( {_id: e.did});
        }
    );

File: Week2/Homework/hw2-2/hw2-2a.rb
#ruby
connection = Mongo::Connection.new
db         = connection.db("students")
grades     = db.collection("grades").find("type": "homework").sort({student_id: 1, score: 1})

student_id = -1
while grades.has_next? do
  grade = grades.next
  if student_id != grade["student_id"]
    student_id = grade["student_id"]
    puts grade
    db.collection("grades").remove(grade)
   end
end
